src/os/bluestore/BlueStore.h

```diff
@@ -2728,4 +2752,149 @@ static inline void intrusive_ptr_release(BlueStore::OpSequencer *o) {
   o->put();
 }
 
+class BlueStoreRepairer
+{
+public:
+  // to simplify future potential migration to mempools
+  using fsck_interval = interval_set<uint64_t>;
+
+  // Structure to track cid/oid -> pextent mapping.
+  // Similar to Bloom filter positive and false-positive matches are 
+  // possible only.
+  // Differs from the BF as matches are verified on CID or OID only.
+  // Allows to filter out 'uninteresting' pextents prior to match verification.
```

Comment:
Note that we have a bloom_filter.hpp that has a method to shrink to a smaller size; can that be reused?